### DISCUSSION TITLE
Allow Starlight plugins to include other plugins

### DISCUSSION THREAD

**Discussion by Fevol** [withastro/starlight#2330] [Feature Requests]
### What version of `starlight` are you using?

0.25.5

### What is your idea?

Expose a `addPlugin` method within the `setup` hook for a plugin, which loads this plugin (immediately) _after_ the current plugin.

### Why is this feature necessary?

In my specific use case, I am attempting to make a theme which includes a separate component defined as a Starlight plugin. While it would technically be possible to add that plugin's integration with the `addIntegration` function, the same is not possible for its custom CSS, overrides, etc.

It _is_ possible to bodge this functionality by calling the plugin's `setup` hook manually, but this brings additional issues with configs only being shallowly merged when both the child and parent plugin call `updateConfig`.

(Example of the bodge)
```ts
export default function plugin(userConfig?: object): StarlightPlugin {
  return {
    name: 'parent',
    hooks: {
      setup: async (args) => {
        const { config, logger, updateConfig } = args;
        if (config.plugins?.some(plugin => plugin.name === 'child')) {
          logger.warn("`child` is already included in the `astro.config.mjs`. Skipping integration.");
        } else {
          await (childPlugin(userConfig).hooks).setup(args);
        }
        
        updateConfig({
          // This will overwrite any component overrides defined by childPlugin
          components: { someComponent: "some-component.astro", ...config.components }
        });
      },
    },
  };
}
```

Additional links of discussions:
- [Previous discussion for adding an `addPlugin` method](https://github.com/withastro/starlight/pull/942#discussion_r1378069549)
- [Discord discussion](https://discord.com/channels/830184174198718474/1070481941863878697/1285727382132490321)

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thanks for the discussion and helpful links @Fevol!

I think having a dedicated `addPlugin()` method in the plugin setup hook could make sense for this as discussed in the original implementation PR you linked.
